src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -461,45 +461,58 @@ private void LogTranscriptHeader(System.Management.Automation.Remoting.PSSenderI
             }
 
             // Add bits from PSVersionTable
-            StringBuilder psVersionInfo = new StringBuilder();
+            StringBuilder versionInfoFooter = new StringBuilder();
             Hashtable versionInfo = PSVersionInfo.GetPSVersionTable();
             foreach (string versionKey in versionInfo.Keys)
             {
-                Object value = versionInfo[versionKey];
+                object value = versionInfo[versionKey];
 
                 if (value != null)
                 {
                     var arrayValue = value as object[];
                     string valueString = arrayValue != null ? string.Join(", ", arrayValue) : value.ToString();
-                    psVersionInfo.AppendLine(versionKey + ": " + valueString);
+                    versionInfoFooter.AppendLine(versionKey + ": " + valueString);
                 }
             }
-            string psConfigurationName = string.Empty;
+
+            string configurationName = string.Empty;
             if (senderInfo != null && !string.IsNullOrEmpty(senderInfo.ConfigurationName))
             {
-                psConfigurationName = senderInfo.ConfigurationName;
+                configurationName = senderInfo.ConfigurationName;
             }
+
             // Transcribe the transcript header
-            string format = InternalHostUserInterfaceStrings.TranscriptPrologue;
-            string line =
-                String.Format(
-                    Globalization.CultureInfo.InvariantCulture,
-                    format,
-                    DateTime.Now,
-                    username,
-                    runAsUser,
-                    psConfigurationName,
-                    Environment.MachineName,
-                    Environment.OSVersion.VersionString,
-                    String.Join(" ", Environment.GetCommandLineArgs()),
-                    System.Diagnostics.Process.GetCurrentProcess().Id,
-                    psVersionInfo.ToString().TrimEnd()
-                    );
+            string line;
+            if (minimalTranscriptHeader)
+            {
+                line =
+                    string.Format(
+                        Globalization.CultureInfo.InvariantCulture,
+                        InternalHostUserInterfaceStrings.MinimalTranscriptPrologue,
+                        DateTime.Now);
```

Comment:
When `minimalTranscriptHeader` is true, there is no need to calculate the unused information like `versioninfo` and `username`. So it's better to organize the code this way:
```
string line;
if (useMinimalHeader)
{
    line = ...
}
else
{
    string username  ...
    string runAsUser ...
    ...
    line = ...
}
```